pkg/kubelet/volume/empty_dir/empty_dir.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package empty_dir
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/volume"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+)
+
+func init() {
+	volume.RegisterPlugin(&emptyDirPlugin{nil, false})
```

Comment:
Concrete example is https://github.com/openshift/geard/blob/master/cmd/gear/linux.go - imagine the kubelet three years from now with 15 different extension types.  Each one has 3-5 implementations.  There are ordering dependencies, where maybe volumes and security need to share data.  All of the sudden, everyone is using static globals for interaction.  On the other hand, if you remove the statics all together, this one file lists everything, and anyone who wants to elide a particular plugin can at any time go remove it in one place (rather than trying to ensure it's not accidentally included somewhere else).